pkg/scheduler/framework/plugins/podtopologyspread/plugin.go

```diff
@@ -291,19 +288,63 @@ func (pl *PodTopologySpread) isSchedulableAfterNodeChange(logger klog.Logger, po
 		return framework.Queue, err
 	}
 
+	var originalNodeMatching, modifiedNodeMatching bool
+	if originalNode != nil {
+		originalNodeMatching = nodeLabelsMatchSpreadConstraints(originalNode.Labels, constraints)
+	}
+	if modifiedNode != nil {
+		modifiedNodeMatching = nodeLabelsMatchSpreadConstraints(modifiedNode.Labels, constraints)
+	}
+
+	// We return Queue in the following cases:
+	// 1. Node/UpdateNodeTaint:
+	//  - Both the original node and the modified node match the pod's topology spread constraints, but the original node and the modified node has different taints.
+	// 2. Node/UpdateNodeLabel:
+	//  - The original node matched the pod's topology spread constraints, but the modified node does not.
+	//  - The modified node matches the pod's topology spread constraints, but the original node does not.
+	//  - Both the original node and the modified node match the pod's topology spread constraints, but their label values are different.
+	// 3. Node/Add: The created node matches the pod's topology spread constraints.
+	// 4. Node/Delete: The original node matched the pod's topology spread constraints.
+
+	if originalNode != nil && modifiedNode != nil {
+		if originalNodeMatching && modifiedNodeMatching && !equality.Semantic.DeepEqual(originalNode.Spec.Taints, modifiedNode.Spec.Taints) {
```

Comment:
Since we have UpdateNodeTaint event, I think when updating Taint, we also need to have `originalNodeMatching && modifiedNodeMatching` conditions.

Reply:
Isn't only `modifiedNodeMatching && !equality.Semantic.DeepEqual(originalNode.Spec.Taints, modifiedNode.Spec.Taints)` enough? 
Why does `originalNodeMatching` have to be true?

Reply:
thanks. fixed